feat: 🎸 missing comment rule

MarcoEidinger opened the pull request:
identify localized strings which do not have a comment in the `.strings`
file

✅ Closes: #15

MarcoEidinger committed:
feat: 🎸 missing comment rule

identify localized strings which do not have a comment in the `.strings`
file

✅ Closes: #15

dral3x commented on Sources/StringsLintFramework/Rules/Lint/MissingCommentRule.swift:
```diff
@@ -0,0 +1,79 @@
+    )
+
+    private var declaredStrings = [LocalizedString]()
+    private var usedStrings = [LocalizedString]()
```
`usedString` is unused, so it can be removed.

dral3x commented on Sources/StringsLintFramework/Rules/Lint/MissingCommentRule.swift:
```diff
@@ -0,0 +1,79 @@
+    var severity: ViolationSeverity
+
+    private let declareParser: LocalizableParser
+    private let usageParser: LocalizableParser
```
`usageParser` is pretty much unused, so it can be removed as well.

dral3x reviewed (changes requested):
Thanks for working on this! I really appreciate it!

I see no change to the `.xcodeproj` file, and when I run the tool via Xcode, the compiler fails. Can you please fix that?

I've also found some possible cleanup to be addressed (see comments below). Nothing major of course, but it's always better not having dead code around.

Thanks!

MarcoEidinger committed:
refactor: 💡 delete unused code

MarcoEidinger committed:
chore: 🤖 add new files to .xcodeproj

building StringsLint in Xcode when opening package by double-clicking on
Package.swift worked fine before. Also building via make install worked
before.

MarcoEidinger commented:
I deleted the unused code.

I added the new files but I could not verify your workflow. I was not able to build with .xcodeproj / workspace even before my changes. Hope by adding the new files it works for you now. If not then I would need your help to understand and run your workflow. I don't know if you expected me to run `swift package generate-xcodeproj.` again. FYI this command is deprecated. Building StringsLint in Xcode when opening package by double-clicking on `Package.swift` worked fine before. Also building via `make install` worked before.

dral3x reviewed (approved):
I didn't know the .xcodeproj was not working for you, sorry about that.

I was not expecting the usage of `generate-xcodeproj` command, but thanks for letting me know about its deprecation!

Anyway, thanks for integrating my requests. Now everything looks awesome! 👏 

I'll merge this PR and release a new version asap.
